parlai/crowdsourcing/tasks/model_chat/analysis/compile_results.py

```diff
@@ -0,0 +1,522 @@
+            self.output_folder,
+            f'results_{"".join([d + "_" for d in date_strings])[:-1]}_{now.strftime("%Y%m%d_%H%M%S")}.csv',
+        )
+        worker_results_file = os.path.join(
+            self.output_folder,
+            f'worker_results_{"".join([d + "_" for d in date_strings])[:-1]}_{now.strftime("%Y%m%d_%H%M%S")}.csv',
+        )
+        # Read in each file
+        num_incomplete_convos = 0
+        num_complete_convos = 0
+        complete_convos_per_model = {}
+        bad_conversations = []
+        old_experimental_design_conversations = []
+        problem_counts = {}
+        worker_stats = {}
+        worker_conversation_counts = {}
+        total_utterances = 0
+
+        conversation_idx = 0
+        conversation_dfs = []
+        for read_folder in read_folders:
+            read_folder_name = os.path.split(read_folder)[-1]
+            for file_name in sorted(os.listdir(read_folder)):
+                if file_name in self.hit_block_list:
+                    continue
+
+                if 'incomplete' in file_name:
+                    num_incomplete_convos += 1
+                    continue
+                else:
+                    num_complete_convos += 1
+
+                # Read in file
+                with open(os.path.join(read_folder, file_name), 'rb') as f:
+                    data = json.load(f)
+
+                # Only include the first max_convos_per_worker conversations from a
+                # worker to avoid biasing
+                worker_id = data['workers'][0]
+                assignment_id = data['assignment_ids'][0]
+                if worker_id in worker_conversation_counts:
+                    conversations_so_far = worker_conversation_counts[worker_id]
+                else:
+                    conversations_so_far = 0
+                worker_conversation_counts[worker_id] = conversations_so_far + 1
+                if (
+                    self.max_convos_per_worker != -1
+                    and conversations_so_far >= self.max_convos_per_worker
+                ):
+                    print(
+                        f'Had {conversations_so_far} conversations already from this worker {worker_id}. Skipping {assignment_id}.'
+                    )
+                    continue
+
+                # Check if need to block the turker
+                word_counts = [
+                    len(d['text'].split(' '))
+                    for d in data['dialog']
+                    if d['agent_idx'] == 0
+                ]
+                utterances = [d['text'] for d in data['dialog'] if d['agent_idx'] == 0]
+                words = [
+                    d['text'].lower().split(' ')
+                    for d in data['dialog']
+                    if d['agent_idx'] == 0
+                ]
+                if np.average(word_counts) < self.min_word_count or (
+                    'contradiction' in self.PROBLEM_BUCKETS and 'contradiction' in words
```

Comment:
can you elaborate what exactly this is checking for? is this saying that the turker marked contradiction in the data and said contradiction?

Reply:
I think it is catching some corner cases where instead to check boxes the turkers just type in their choices. This is very specific to this particular q function task tho I guess?

Reply:
Hmm, fair enough, it *is* very specific to one particular Q-function task. I was going for feature completeness in this port, but such a specific check doesn't belong in a general analysis script. Removed